json/json.go

```diff
@@ -170,12 +170,14 @@ func Unescape(s []byte) []byte {
 }
 
 // AppendUnescape appends s to b with the string unescaped as a JSON value.
-// This will include the starting and ending quote characters, and the
+// This will remove starting and ending quote characters, and the
 // appropriate characters will be escaped correctly as if JSON decoded.
+// New space will be reallocated if more space is needed.
 func AppendUnescape(b []byte, s []byte, flags ParseFlags) []byte {
 	d := decoder{flags: flags}
-	b, _ = d.decodeString(s, unsafe.Pointer(&s))
-	return s
+	buf := new(string)
+	d.decodeString(s, unsafe.Pointer(buf))
```

Comment:
I'm not super familiar with this code, so bear with me here. 😁  Is it possible to decode straight into `b` to avoid allocating to `buf` and then copying?

Reply:
unlike `encodeString()`, `decodeString()` does not append the result string to the buffer passed it, rather it simply overwrites the string at the [pointer passed in](https://github.com/segmentio/encoding/blob/master/json/decode.go#L292):
`*(*string)(p) = string(s)`, where p is the passed in buffer. I'd like to avoid modifying the underlying function for this simple change, but perhaps there are options I am be missing!

Reply:
Maybe using `decodeString` isn't the right approach, have you looked into [`parseStringUnquote`](https://github.com/segmentio/encoding/blob/master/json/parse.go#L440:6) ?

Reply:
I guess you'd have to reimplement the handling of parsing flag, maybe that's not the right path.